java/ql/test/library-tests/dataflow/collections/Test.java

```diff
@@ -84,7 +84,17 @@ public void run4() {
     p.put("key", tainted);
     sink(p.getProperty("key")); // Flow
     sink(p.getProperty("key", "defaultValue")); // Flow
-    
+
+    Properties clean = new Properties();
+    sink(clean.getProperty("key", tainted)); // Flow
+  }
+
+  public void run5() {
+    Properties p = new Properties();
+    p.setProperty("key", tainted);
```

Comment:
You should test that the return value of setProperty returns taint from the map too

Reply:
Done in https://github.com/github/codeql/pull/9963/commits/8eebe710ce568c49f57c51d958b211e4e0e4fa63